### PR TITLE
Relate next

### PR DESCRIPTION
**Pull request by gregghz**
- Fixes https://github.com/lucidsoftware/relate/issues/6
- Repurposes the deprecated `RowParser` as the new name for `Parseable` (I don't like that name anymore)
- Removes things like `SQL()` and related code (I may have missed some)
- Moves everything in `com.lucidchart.open.relate.interp` into `com.lucidchart.open.relate`. Having a package specifically for interpolated queries isn't necessary as I've removed all other methods of writing queries.

This branch is more or less a WIP for a 2.0.0 release (since it removes a lot of the API and replaces many instances of `SqlResult` with `SqlRow`).

### CODE REVIEW

**Comment by gregghz** [comment]
I haven't updated the regression tests or the benchmarks yet.

File: relate/src/main/scala/com/lucidchart/open/relate/SqlRow.scala [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+}
+
+class SqlRow(val resultSet: java.sql.ResultSet) extends ResultSetWrapper {
+  /**
+    * Get the number of the row the SqlResult is currently on
+    * @return the current row number
+    */
+  def getRow(): Int = resultSet.getRow()
+
+  def strictArray(column: String): java.sql.Array = resultSet.getArray(column)
```
Comment (msiebert) [inline relate/src/main/scala/com/lucidchart/open/relate/SqlRow.scala]: whew, when I saw you'd deleted the strict* methods in the other file, I was worried that they were gone forever. Then what would I not use?

**Comment by msiebert** [comment]
These changes look good to me. I'm really glad you got rid of the old style RowParsers

File: relate/src/main/scala/com/lucidchart/open/relate/SqlRow.scala (hunk 1 above)
Comment (gregghz) [inline relate/src/main/scala/com/lucidchart/open/relate/SqlRow.scala]: well honestly, I'm more inclined to remove the non-strict versions and replace them with the strict versions.

**Comment by gregghz** [comment]
@msiebert interestingly, the "new" RowParser is pretty much the same as the old one. The only difference is that it's not covariant and it doubles as a type class providing parseability.

In fact, I think I'm going to make it not extend `SqlRow => A`

**Comment by pauldraper** [comment]
If you're going to rename `com.lucidchart.open.relate`, just go whole hog and name it `com.lucidchart.relate`.

**Comment by msiebert** [comment]
That or `co.golucid.relate` :stuck_out_tongue:

**Pull request merged**